app/src/main/java/org/mozilla/fenix/browser/MainOnboardingObserver.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.browser
+
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.state.state.SessionState
+import org.mozilla.fenix.onboarding.FenixOnboarding
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * Dismisses onboarding when user navigates to a page, other than the links from onboarding screen,
+ * nominated in [onboardingLinksList]
+ */
+class MainOnboardingObserver(
+    private val onboarding: FenixOnboarding
+) : Session.Observer {
```

Comment:
We are trying to not use `Session.Observer` unless is strictly necessary as we are going to deprecated it soon, lets use the `BrowserStore` instead, we could do it in two ways:

1. Creating an `OnboardingMiddleware` similar to [RecentlyClosedMiddleware](https://github.com/mozilla-mobile/android-components/blob/5451ed708c21750a18c207a6ecae3972aa9aa839/components/feature/recentlyclosed/src/main/java/mozilla/components/feature/recentlyclosed/RecentlyClosedMiddleware.kt#L31) and checking by the `UpdateLoadRequestAction`

2. Observing the `content.loadRequest` via flow you can find an example [here](https://github.com/mozilla-mobile/fenix/commit/d7688c84278b74e4d96688c3c223cbfa14b4885b)